schemas/stsci.edu/asdf/0.1.0/asdf-schema.yaml

```diff
@@ -0,0 +1,46 @@
+%YAML 1.1
+---
+$schema: "http://json-schema.org/draft-04/schema"
+id: "http://stsci.edu/schemas/asdf/0.1.0/asdf-schema"
+title:
+  ASDF schema
+description: |
+  A metaschema extending YAML Schema and JSON Schema to add support
+  for some ASDF-specific checks, related to nd-arrays.
```

Comment:
The layering of metaschemas I understood.

I guess what I'm confused by (and I didn't sleep well last night so I'm probably off-focus) is the need for special properties for this.  For example ndarray objects already have a shape property.  So is not possible for a schema to require something like:
```yaml
tag: ndarray
properties:
    shape:
        minItems: 2
        maxItems: 2
```

(or something to that effect; I forget the exact syntax).

That covers both `ndim` and `max_ndim`.  Seems like you could do something similar with datatype.

Anyways that said, having these special keywords is more convenient and I'm not against it.  I'm just trying to make sure I understand the thinking behind it.